hnn_core/opt_toy_example/general.py

```diff
@@ -0,0 +1,370 @@
+import numpy as np
+
+from hnn_core.network import pick_connection
+
+from metrics import _rmse_evoked, _rmse_rhythmic, _rmse_poisson
+
+from skopt import gp_minimize
+from scipy.optimize import fmin_cobyla
+
+
+class Optimizer:
+    def __init__(self, net, constraints, solver, obj_fun, f_bands=None,
+                 weights=None):
+        self.net = net
+        self.constraints = constraints
+        # Optimizer method
+        if solver == 'bayesian':
+            self._get_params = _get_params_bayesian
+            self._run_opt = _run_opt_bayesian
+        elif solver == 'cobyla':
+            self._get_params = _get_params_cobyla
+            self._run_opt = _run_opt_cobyla
+        # Response to be optimized
+        if obj_fun == 'evoked':
+            self.obj_fun = _rmse_evoked
+        elif obj_fun == 'rhythmic':
+            self.obj_fun = _rmse_rhythmic
+            self.f_bands = f_bands
```

Comment:
Add logic t check if `f_bands` is in the appropriate format

Add warning otherwise